pages/deployments/deploying_to_kubernetes.md.erb

```diff
@@ -0,0 +1,222 @@
+queue.
+
+## Writing the Deploy Script
+
+The next step is writing a deploy script that generates a [Kubernetes deployment
+manifest][k8s_deployment] from the `DOCKER_IMAGE` environment variable.
+
+Let's start with manifest file. This sample file creates a Deployment with
+three replicas (horizontal scale in Kubernetes lingo) each listening port
+`3000`. Change the `containerPort` to fit your application.
+
+<section class="Docs__note">
+  <p>
+    The <a
+    href="https://kubernetes.io/docs/concepts/workloads/controllers/deployment/">official
+    deployment documentation</a> covers much more than what fits in this
+    tutorial. Refer back to these docs for information on setting CPU and
+    memory, controlling networking, deployment update strategies, and how to
+    expose your application to the internet.
+  </p>
+</section>
+
+Let's call this file `k8s/deployment.yml`.
+
+```yml
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: tutorial
+  labels:
+    app: tutorial
+spec:
+  # TODO: replace with a value that fits your application
+  replicas: 3
+  selector:
+    matchLabels:
+      app: tutorial
+  template:
+    metadata:
+      labels:
+        app: tutorial
+    spec:
+      containers:
+        - name: app
+          image: "${DOCKER_IMAGE}"
+          ports:
+            # TODO: replace with the correct port for your application
+            - containerPort: 3000
+```
+
+Note manifest includes `${DOCKER_IMAGE}`. There is no environment variable
+subsitution in YAML or `kubectl` itself. This is where the deploy script comes
+in. The deploy script can use `envsubst` ("environment subsitute";
+[docs](https://linux.die.net/man/1/envsubst)) as a minimal templating solution.
+The resulting output may be piped directly into `kubectl`.
+
+The full script has three parts:
+
+1. Check `$DOCKER_IMAGE` is set
+2. Generate a complete manifest with `envsubt` and apply with `kubectl`
+3. Wait for Kubernetes to complete the deploy.
+
+This fits neatly into a Bash script. Here's the complete `script/buildkite/deploy`:
+
+```bash
+#!/usr/bin/env bash
+
+set -euo pipefail
+
+if [ -z "${DOCKER_IMAGE:-}" ]; then
+  echo "\:boom\: \$DOCKER_IMAGE missing" 1>&2
+  exit 1
+fi
+
+echo '--- \:kubernetes\: Shipping'
+envsubst < k8s/app.yml | kubectl apply -f -
+
+
+echo '--- \:zzz\: Waiting for rollout'
+# TODO: add the wait step
+```
+
+You can test pipeline now that everything is in place. All you need is your
+Docker image.
+
+## Test the Pipeline
+
+Open the deployment pipline and click "New Build". Click "Options" and set the
+`DOCKER_IMAGE` environment variable.
+
+<%= image "manual_deployment.png", width: 585/2, height: 661/2, alt: "New Manual Build" %>
+
+Assuming your agents have the required access, then success! :tada:
+
+<%= image "manual_deployment_success.png", width: 743/2, height: 728/2, alt: "Manual Build Success" %>
+
+## Automate Deployments
+
+We can do better than manual deployments. We'll use a [trigger
```

Comment:
This first line feels a bit too direct/blunt? If that makes sense? Maybe something as subtle as adding '- continuous deployments' after? Or replace it with something more basic like 'Next we'll set up continuous deploys'?